sdk/documenttranslation/Azure.AI.DocumentTranslation/src/ClientDiagnostics.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Text.Json;
+using Azure.AI.DocumentTranslation;
+
+#nullable enable
+
+namespace Azure.Core.Pipeline
+{
+    /// <summary>
+    /// Extend ClientDiagnostics to customize the exceptions thrown by the
+    /// generated code.
+    /// </summary>
+    internal sealed partial class ClientDiagnostics
+    {
+        /// <summary>
+        /// Customize the exception messages we throw from the protocol layer by
+        /// attempting to parse them as <see cref="DocumentTranslationError"/>s.
+        /// </summary>
+        /// <param name="content">The error content.</param>
+        /// <param name="responseHeaders">The response headers.</param>
+        /// <param name="message">The error message.</param>
+        /// <param name="errorCode">The error code.</param>
+        /// <param name="additionalInfo">Additional error details.</param>
+#pragma warning disable CA1822 // Member can be static
+#pragma warning disable CA1801 // Remove unused parameter
+        partial void ExtractFailureContent(
+            string? content,
+            ResponseHeaders responseHeaders,
+            ref string? message,
+            ref string? errorCode,
+            ref IDictionary<string, string>? additionalInfo)
+#pragma warning restore CA1801 // Remove unused parameter
+#pragma warning restore CA1822 // Member can be static
+        {
+            if (!string.IsNullOrEmpty(content))
+            {
+                try
+                {
+                    // Try to parse the failure content and use that as the
+                    // default value for the message, error code, etc.
+                    using JsonDocument doc = JsonDocument.Parse(content);
+                    if (doc.RootElement.TryGetProperty("error", out JsonElement errorElement))
+                    {
+                        DocumentTranslationError error = DocumentTranslationError.DeserializeDocumentTranslationError(errorElement);
+                        message ??= error?.Message;
```

Comment:
If both `message` and `errorCode` are already assigned, maybe skip all this extra work and allocation since you're conditionally setting them if null anyway. Given how/when this method is called, unnecessary work should be elided.

Reply:
Good point.